gridformat/traits/mfem.hpp

```diff
@@ -0,0 +1,138 @@
+// SPDX-FileCopyrightText: 2022 Dennis Gläser <dennis.glaeser@iws.uni-stuttgart.de>
+// SPDX-License-Identifier: GPL-3.0-or-later
+/*!
+ * \file
+ * \ingroup PredefinedTraits
+ * \brief Traits specializations for
+ *        <a href="https://docs.mfem.org/html/classmfem_1_1Mesh.html">mfem::Mesh</a>
+ */
+#ifndef GRIDFORMAT_TRAITS_MFEM_HPP_
+#define GRIDFORMAT_TRAITS_MFEM_HPP_
+
+#include <cassert>
+#include <ranges>
+#include <vector>
+#include <array>
+#include <type_traits>
+
+#include <gridformat/common/exceptions.hpp>
+#include <gridformat/common/filtered_range.hpp>
+
+#include <gridformat/grid/cell_type.hpp>
+#include <gridformat/grid/traits.hpp>
+
+// forward declaration of the triangulation classes
+// users are expected to include the headers of the actually used classes
+namespace mfem {
+
+// TODO
+// #ifdef MFEM_USE_MPI
```

Comment:
I guess the `ifdef MFEM_USE_MPI` is needed because if mpi is not found, the class `ParMesh` is not "visible"?

See #138. I thought it could be good to search for packages for which there exist predefined traits when configuring `GridFormat`, we could then define respective `GRIDFORMAT_HAVE_PKG_XYZ`. This way, we could

- include the headers for the types we are using ourselves - so users don't have to include stuff in a particular order
- guard specific features of a package behind those variables (as this one right here, for instance)

I can set that up, then we can make use of it here.

Reply:
are the guards needed to include the headers? I thought the examples are only compiled if the libs are found anyway. For a user it also doesn’t matter because they have their lib already set up and then inlcude the corresponding traits header.

Reply:
yeah maybe they are not needed. I just thought we could also check for versions and then have traits specializations depending on the versions. In any case, I would prefer to include the headers over the (current) forward declarations.

Reply:
yes I think including the headers is better. 

and yes version macros would make sense if the libraries don’t already provide them.